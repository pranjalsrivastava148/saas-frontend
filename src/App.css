.container {
  background-color: #16181b;
  height: 100vh;
  padding: 36px;
  overflow-y: hidden;
}
