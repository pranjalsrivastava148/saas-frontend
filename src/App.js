import { useEffect, useState } from "react";
import "./App.css";
import Table from "./components/Table/Table";

const LINK_TO_API =
  "https://raw.githubusercontent.com/saaslabsco/frontend-assignment/refs/heads/master/frontend-assignment.json";

const columns = [
  { headerName: "S.No.", field: "s.no" },
  { headerName: "Percentage Funded", field: "percentage.funded" },
  { headerName: "Amount Pledged", field: "amt.pledged" },
];

function App() {
  const [projects, setProjects] = useState([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  const sortData = (data) => {
    return [...data].sort((a, b) => {
      if (a["percentage.funded"] < b["percentage.funded"]) {
        return 1;
      }
      if (a["percentage.funded"] > b["percentage.funded"]) {
        return -1;
      }
      return 0;
    });
  };

  useEffect(() => {
    const fetchProjects = async () => {
      try {
        const response = await fetch(LINK_TO_API);
        if (response) {
          const data = await response.json();
          // const highlyRatedProjects = sortData(data);
          setProjects(data);
        }
      } catch (err) {
        setError(err.message);
      } finally {
        setLoading(false);
      }
    };

    fetchProjects();
  }, []);

  if (loading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return <div>Error: {error}</div>;
  }

  return (
    <div className="container">
      <Table
        data={projects}
        columns={columns}
        rowsPerPage={5}
      />
    </div>
  );
}

export default App;
