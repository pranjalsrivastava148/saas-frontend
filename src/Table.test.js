import { render, screen } from '@testing-library/react';
import App from './App';

global.fetch = jest.fn(() =>
  Promise.resolve({
    ok: true,
      json: jest.fn().mockResolvedValueOnce([
        { id: 1, title: "Project 1" },
        { id: 2, title: "Project 2" },
        { id: 3, title: "Project 3" },
      ]),
  })
);

describe('Table with data', () => {
  beforeEach(() => {
    fetch.mockClear();
  });

  it('renders learn react link', () => {
    fetch.mockResolvedValueOnce({
      ok: true,
      json: jest.fn().mockResolvedValueOnce([
        { id: 1, title: "Project 1" },
        { id: 2, title: "Project 2" },
        { id: 3, title: "Project 3" },
      ]),
    });

    const wrapper = render(<App />);
    screen.debug()
    const linkElement = screen.getByText(/loading.../i);
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toMatchSnapshot();
  });
})

