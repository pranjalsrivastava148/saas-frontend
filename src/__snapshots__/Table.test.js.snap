// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table with data renders learn react link 1`] = `
<div>
  Loading...
</div>
`;
